### PR TITLE
Create stories for profile search

### PR DESCRIPTION
**Pull request by akalin-keybase**
Also create selectorMap functions for various search components.

Improve typing here and there.

### CODE REVIEW

**Comment by akalin-keybase** [comment]
r? @keybase/react-hackers 

After doing a bunch of stories cleanup, I feel like a connected component should include a story using its connected variant, so that all the property-mocking logic lives close to the component code, and that logic should be exposed for users of the connected component. Thoughts?

**Comment by akalin-keybase** [comment]
Rebased on master. Ping!

File: shared/search/result-row/container.js [hunk 1]
```diff
@@ -1,38 +1,37 @@
...
   onShowTracker?: () => void,
 }
 
-const mapStateToProps = (
-  state: TypedState,
-  {disableIfInTeamName, id, onClick, onMouseOver, onShowTracker}: OwnProps
-) => {
-  const result: any = state.entities.getIn(['search', 'searchResults', id], Map()).toObject()
+const mapStateToProps = (state: TypedState, ownProps: OwnProps) => {
+  const result = state.entities.search.searchResults.get(ownProps.id, makeSearchResult()).toObject()
```
Comment (chrisnojima) [inline shared/search/result-row/container.js]: i know this is old code but we should never do this. do this in mergeprops

Comment (akalin-keybase) [reply shared/search/result-row/container.js]: Done

**Comment by akalin-keybase** [comment]
PTAL!

File: shared/search/result-row/container.js [hunk 2]
```diff
@@ -1,42 +1,54 @@
...
-}
+|}
 
-const mapStateToProps = (
-  state: TypedState,
-  {disableIfInTeamName, id, onClick, onMouseOver, onShowTracker}: OwnProps
-) => {
-  const result: any = state.entities.getIn(['search', 'searchResults', id], Map()).toObject()
+const mapStateToProps = (state: TypedState, ownProps: OwnProps) => {
+  const result = state.entities.search.searchResults.get(ownProps.id, makeSearchResult())
```
Comment (chrisnojima) [inline shared/search/result-row/container.js]: should make this a constant variable else it'll make a new record each time and render thrash
aka
const emptySearch = makeSearchResult() // in the module
searchResults.get(ownProps.id, emptySearch)

Comment (akalin-keybase) [reply shared/search/result-row/container.js]: d'oh, done.

**Comment by akalin-keybase** [comment]
PTAL @chrisnojima

**Comment by akalin-keybase** [comment]
circleci's kinda backed up, and a previous run succeeded, so merging

**Pull request merged**